cmake/pip_utils.cmake

```diff
@@ -0,0 +1,164 @@
+    else()
+        set(HTML_DOC_DEPENDENCY_OPTION ${PIP_ADD_HTML_TARGET_DEPENDS})
+    endif()
+    set(${PIP_ADD_HTML_TARGET_NAME} ${HTML_DOC_DEPENDENCY_OPTION} PARENT_SCOPE)
+endfunction()
+
+function(pip_add_dist_target)
+    set(options)
+    set(oneValueArgs NAME)
+    set(multiValueArgs DEPENDS)
+    cmake_parse_arguments(PIP_ADD_DIST_TARGET "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    add_custom_command(
+        OUTPUT ${WORKING_DIR}/dist.md5
+        DEPENDS ${PIP_ADD_DIST_TARGET_DEPENDS}
+        COMMAND /usr/bin/env python3 setup.py bdist bdist_wheel sdist
+        COMMAND tar c ${PYTHON_PACKAGE_DIR}/dist 2> /dev/null | md5sum > ${WORKING_DIR}/dist.md5
+        WORKING_DIRECTORY ${PYTHON_PACKAGE_DIR}
+        VERBATIM
+    )
+    add_custom_target(
+        ${PIP_ADD_DIST_TARGET_NAME}-target
+        ALL
+        DEPENDS ${WORKING_DIR}/dist.md5
+        VERBATIM
+    )
+    set(${PIP_ADD_DIST_TARGET_NAME} ${PIP_ADD_DIST_TARGET_NAME}-target ${WORKING_DIR}/dist.md5 PARENT_SCOPE)
+endfunction()
+
+function(pip_add_install_target)
+    set(options)
+    set(oneValueArgs NAME INSTALL_ON_BUILD PIP_DEVEL_PREFIX PIP_INSTALL_PREFIX)
+    set(multiValueArgs DEPENDS)
+    cmake_parse_arguments(PIP_ADD_INSTALL_TARGET "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    if(PIP_ADD_INSTALL_TARGET_INSTALL_ON_BUILD)
+        set(ALL_OPTION ALL)
+    else()
+        set(ALL_OPTION)
+        file(REMOVE ${WORKING_DIR}/install.md5)
+    endif()
+
+    configure_file(
+        ${CMAKE_UTILS_DIR}/pip_install.cmake.in
+        ${WORKING_DIR}/pip_install.cmake
+        @ONLY
+    )
+
+    add_custom_command(
+        OUTPUT ${WORKING_DIR}/install.md5
+        DEPENDS ${PIP_ADD_INSTALL_TARGET_DEPENDS} ${WORKING_DIR}/pip_install.cmake
+        COMMAND ${CMAKE_COMMAND} -DPIP_INSTALL_PREFIX=${PIP_ADD_INSTALL_TARGET_PIP_DEVEL_PREFIX} -P ${WORKING_DIR}/pip_install.cmake
+        WORKING_DIRECTORY ${PYTHON_PACKAGE_DIR}
+        VERBATIM
+    )
+    add_custom_target(
+        ${PIP_ADD_INSTALL_TARGET_NAME}-target
+        ${ALL_OPTION}
+        DEPENDS ${WORKING_DIR}/install.md5
+        VERBATIM
+    )
+    set(${PIP_ADD_INSTALL_TARGET_NAME} ${PIP_ADD_INSTALL_TARGET_NAME}-target ${WORKING_DIR}/install.md5 PARENT_SCOPE)
+    install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -DPIP_INSTALL_PREFIX=${PIP_ADD_INSTALL_TARGET_PIP_INSTALL_PREFIX} -P ${WORKING_DIR}/pip_install.cmake)")
+endfunction()
+
+#######################
+## Required programs ##
+#######################
+function(assert_program_installed PROGRAM)
+    find_program(prog ${PROGRAM})
+    if(${prog} STREQUAL "prog-NOTFOUND")
+        message(FATAL_ERROR "'${PROGRAM}' needs to be installed")
+    endif()
+endfunction()
+
+assert_program_installed("perl")
+assert_program_installed("python3")
+assert_program_installed("rsync")
+assert_program_installed("tar")
+assert_program_installed("md5sum")
```

Comment:
Explicit check for programs needed in `add_custom_command` and `execute_process` and all